modules/katello_host_collection.py

```diff
@@ -0,0 +1,130 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_host_collection
+short_description: Create and Manage host collections
+description:
+    - Create and Manage host collections
+author:
+    - "Maxim Burgerhout (@wzzrd)"
+requirements:
+    - "nailgun >= 0.32.0"
+    - "python >= 2.6"
+    - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    default: true
+    type: bool
+  description:
+    description:
+      - Description of the host collection
+    required: false
+  organization:
+    description:
+      - Organization that the host collection is in
+    required: true
+  name:
+    description:
+      - Name of the host collection
+    required: true
+  state:
+    description:
+      - State of the host collection
+    default: present
+    choices:
+      - present
+      - absent
+      - present_with_defaults
+'''
+
+EXAMPLES = '''
+- name: "Create Foo host collection"
+  katello_host_collection:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Foo"
+    description: "Foo host collection for Foo servers"
+    organization: "My Cool new Organization"
+    state: present
+'''
+
+RETURN = ''' # '''
+
+try:
+    from ansible.module_utils.foreman_helper import (
+        sanitize_entity_dict,
+    )
+except ImportError:
+    pass
+
+from ansible.module_utils.foreman_helper import KatelloEntityApypieAnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'organization': 'organization',
+    'description': 'description',
+}
+
+
+def main():
+    module = KatelloEntityApypieAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            organization=dict(required=True),
+            description=dict(),
+        ),
+        supports_check_mode=True,
+    )
+
+    (entity_dict, state) = module.parse_params()
+
+    module.connect()
+
+    entity_dict['organization'] = module.find_resource_by_name('organizations', entity_dict['organization'], thin=True)
```

Comment:
I believe, searching the organization is already done by `Katello.*Module`.

Reply:
Ok, removing that line then.

Reply:
Hmm, not as such. But it is added to the argument_spec for you:
https://github.com/theforeman/foreman-ansible-modules/blob/master/module_utils/foreman_helper.py#L292

Reply:
Oh, I should remove the line in the `argument_spec` then.

Reply:
:+1: exactly. I thouhgt i saw something, and it was the argument_spec.